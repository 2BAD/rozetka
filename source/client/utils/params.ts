import { OptionsOfTextResponseBody as Params } from 'got'

/**
 * Create Options object with searchParams populated
 */
export const params = (params: Record<string, string>): Params => ({
  searchParams: new URLSearchParams(params)
})
