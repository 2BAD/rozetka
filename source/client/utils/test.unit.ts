import { params } from './params'

describe('parameter builder', () => {
  it('should return Options object with searchParam set to URLSearchParams object instance', () => {
    expect.assertions(3)

    const p = params(['goodsId', '100'])

    expect(p).toHaveProperty('searchParams')
    expect(p.searchParams).toBeInstanceOf(URLSearchParams)
    expect(p).toMatchSnapshot()
  })

  it('should accept an array of arrays', () => {
    expect.assertions(2)

    const p = params(['goodsId', '100'], ['lang', 'ru'])

    expect(p.searchParams?.toString()).toBe('goodsId=100&lang=ru')
    expect(p).toMatchSnapshot()
  })
})
