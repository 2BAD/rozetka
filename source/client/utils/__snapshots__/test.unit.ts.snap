// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parameter builder should accept an object with records 1`] = `
Object {
  "searchParams": URLSearchParams {
    Symbol(query): Array [
      "goodsId",
      "100",
      "lang",
      "ru",
    ],
    Symbol(context): null,
  },
}
`;

exports[`parameter builder should return Options object with searchParam set to URLSearchParams object instance 1`] = `
Object {
  "searchParams": URLSearchParams {
    Symbol(query): Array [
      "goodsId",
      "100",
    ],
    Symbol(context): null,
  },
}
`;
